compendium/DeclarativeServices/test/TestConfigPolicies.cpp

```diff
@@ -0,0 +1,340 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "gtest/gtest.h"
+#include "TestFixture.hpp"
+
+#include "TestInterfaces/Interfaces.hpp"
+
+#include "cppmicroservices/servicecomponent/detail/ComponentInstanceImpl.hpp"
+
+namespace test {
+
+TEST_F(tServiceComponent, testOptionalConfigPolicyImmediateTrue) // DS_CAI_FTC_5
```

Comment:
Not sure we're testing the right thing here. The test procedure says "Verify that the component can be satisfied even if the corresponding configuration object is not present and can use corresponding configuration object if present." 

So basically there should be two tests. One should show that if the configuration object is not present when the component instance is constructed, the component is constructed with no properties. The second test should show that if the configuration object is present when the component instance is constructed, it should be constructed with the properties. I would use immediate = true, configuration-policy="optional"  for both cases.
 
Test1 - Construct an instance once with no properties available. Use GetProperties to show that the properties in the component instance have default values. 
Test2 - Construct an instance the second time with a configuration object available. Use GetProperties to show that the component instance was constructed with the updated properties from the configuration object.